build: Update intersphinx mapping

sarina opened the pull request:
@feanil - I saw you did this for the XBlock docs, so I figured I should do it for other docs that have moved. However I'm not sure what else I need to do to effect redirects. Note we are also currently in the process of moving the OLX docs (https://github.com/openedx/docs.openedx.org/pull/819)

feanil reviewed (commented):
You shouldn't point to the index.html but to the `en/latest` url because intersphinx builds a URL on top of that to find the inventory files that are generated as a part of docs builds.

sarina commented:
So, I'm getting a million build failures related to missing labels, and I'm going to assume a part of that is that when we moved docs, we changed a number of the labels....

sarina commented:
@feanil this is ready to go. I'll open up separate PRs to figure out the redirects for the open edx course staff docs and eventually the OLX docs once https://github.com/openedx/docs.openedx.org/pull/819 is merged

feanil commented on shared/conf.py:
```diff
@@ -209,19 +209,19 @@ def ism_location(dir_name):
 
 intersphinx_mapping = {
-    "opencoursestaff" : (openedx_rtd_url("open-edx-building-and-running-a-course"), ism_location("open_edx_course_authors")),
+    "opencoursestaff" : ("https://docs.openedx.org/en/latest/", None),
```
FYI, this is because you set the second parameter to `None` that you had to update all the prefixes.  Not something you need to fix but for future reference.
